frameworks/CSharp/appmpower/src/HttpApplication.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting.Server;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using appMpower.Kestrel;
+using PlatformBenchmarks;
+
+namespace appMpower
+{
+   public class HttpApplication : IHttpApplication<IFeatureCollection>
+   {
+      private readonly static AsciiString _plainText = "Hello, World!";
+
+      public IFeatureCollection CreateContext(IFeatureCollection featureCollection)
+      {
+         return featureCollection;
+      }
+
+      public async Task ProcessRequestAsync(IFeatureCollection featureCollection)
+      {
+         var request = featureCollection as IHttpRequestFeature;
+         var httpResponse = featureCollection as IHttpResponseFeature;
+         var httpResponseBody = featureCollection as IHttpResponseBodyFeature;
+
+         PathString pathString = request.Path;
+
+         if (pathString.HasValue)
+         {
+            int pathStringLength = pathString.Value.Length;
+            string pathStringStart = pathString.Value.Substring(1, 1);
+
+            if (pathStringLength == 10 && pathStringStart == "p")
+            {
+               PlainText.Render(httpResponse.Headers, httpResponseBody.Writer, _plainText);
+               return;
+            }
+            else if (pathStringLength == 5 && pathStringStart == "j")
+            {
+               Json.RenderOne(httpResponse.Headers, httpResponseBody.Writer, new JsonMessage { message = "Hello, World!" }, new JsonMessageSerializer());
+               return;
+            }
+            else if (pathStringLength == 3 && pathStringStart == "d")
+            {
+               Json.RenderOne(httpResponse.Headers, httpResponseBody.Writer, await RawDb.LoadSingleQueryRow(), new WorldSerializer());
```

Comment:
`new WorldSerializer()` also looks idempotent and everything is passed via parameter; so could hoist this class to a readonly static rather than allocating each time?